src/test/java/org/apache/commons/pool2/impl/TestGenericKeyedObjectPool.java

```diff
@@ -2474,6 +2474,138 @@ public void testWhenExhaustedBlockClosePool() throws Exception {
         assertTrue(wtt.thrown instanceof InterruptedException);
     }
 
+    /**
+     * POOL-391
+     * Adapted from code in the JIRA ticket.
+     *
+     * @throws Exception May occur in some failure modes
+     */
+    @Test
```

Comment:
Great job on using the code from the JIRA for the test :tada: 

Unfortunately I think there is something wrong with the formatting here :point_down: